### DISCUSSION TITLE
managing legacy and new layouts in fumadocs-ui

### DISCUSSION THREAD

**Discussion by ryanwaits** [fuma-nama/fumadocs#535] [Docs Feedback]
With the release of fumadocs-ui v12.3.0, there were major changes introduced that significantly impact the layout and component structure of the documentation UI: mainly the Navbar has been removed and replaced with a sidebar.

Ideally we would like to upgrade the `ui` package without the required layout change, are there any efforts to make this possible? Or any recommended workarounds?

---

_I attempted one workaround by aliasing the package versions in package.json to use different versions in different parts of the application. This didn't quite work out of the box as it introduces complexity and dependency management issues._

Appreciate the feedback!

**Comment by fuma-nama** [comment]
Actually this is a change of v12, that's why it's marked as a major upgrade. I'm not planning to leave legacy layouts to the package since they may no longer compatible with newer core packages, and I want to reduce the efforts to maintain Fumadocs.

The previous versions (v11) should work fine on production environments, welcome to open a question in discussion if you have any problems during the migration.
